src/main/java/net/coderbot/iris/uniforms/CommonUniforms.java

```diff
@@ -70,4 +102,31 @@ private static int isEyeInWater() {
 			return 0;
 		}
 	}
+
+	private static class HeldItemLightingSupplier implements IntSupplier {
+
+		private final Hand hand;
+
+		private HeldItemLightingSupplier(Hand targetHand) {
+			this.hand = targetHand;
+		}
+
+		@Override
+		public int getAsInt() {
+			if (client.player == null) {
```

Comment:
This is fine as-is, but I do wonder whether it would be better to use the camera entity instead? It's an open question, since that would only have an effect in spectator mode in rare circumstances.

Reply:
For that uniform specifically, it probably wouldn't apply to spectators since it's for the item the player is holding, which in spectator should be nothing.

Reply:
However for other uniforms, it probably should be kept as camera entity then so it gets updated for stuff like playerMood